client-v2/src/actions/Collections.ts

```diff
@@ -101,6 +102,37 @@ function updateCollection(collection: Collection): ThunkResult<Promise<void>> {
   };
 }
 
+/**
+ * Fetch articles from CAPI and add them to the store.
+ */
+const fetchArticles = (articleIds: string[]) => async (dispatch: Dispatch) => {
+  const articleIdsWithoutSnaps = articleIds.filter(id => !id.match(/^snap/));
+  if (!articleIdsWithoutSnaps.length) {
+    return;
+  }
+  dispatch(externalArticleActions.fetchStart(articleIdsWithoutSnaps));
+  try {
+    const articles = await getArticles(articleIdsWithoutSnaps);
+    dispatch(externalArticleActions.fetchSuccess(articles));
+    const remainingArticles = difference(
+      articleIdsWithoutSnaps,
+      articles.map(_ => _.id)
+    );
+    if (remainingArticles.length) {
+      dispatch(
+        externalArticleActions.fetchError(
+          `The following article ids were in a CAPI query but were not returned by CAPI: ${remainingArticles.join(
```

Comment:
This is not for now but when we start looking into error handling in the tool, we need to think about errors which we show to the user and errors which we can see if this happens.

Reply:
Absolutely.